lib/orogen/gen/tasks.rb

```diff
@@ -306,6 +306,48 @@ def register_for_generation
             # <tt>_time</tt> to be added to the generated class (more specifically,
             # to the +Base+ subclass).
             module TaskContextGeneration
+                def initialize(project, name = nil, *args, **options)
+                    RTT_CPP.verify_valid_identifier(name, "task")
+                    super
```

Comment:
since it is a module, to whom is super pointing?

Reply:
It is pre-pended in a class (placing the methods above the class). See `lib/orogen/gen/enable`

It would not be needed anymore in the current structure - i.e. we could have proper subclasses, but refactoring the codegen part of orogen is low on the priority list.